### PR TITLE
[Docker] Add dimensions to metrics data streams, except for events

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Add dimensions to metrics data streams, except for events.

## Details

For all data streams, there exists these set of dimensions:
```
- external: ecs
  name: cloud.account.id
  dimension: true
- external: ecs
  name: cloud.provider
  dimension: true
- external: ecs
  name: cloud.region
  dimension: true
- external: ecs
  name: cloud.availability_zone
  dimension: true
- external: ecs
  name: cloud.instance.id
  dimension: true
```

In case it is running on cloud.More info on this can be found in https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359.

The dimensions set for each data stream are `container.id` and `service.address`, since the ID is unique per Docker daemon. Some data streams did not hold value for `container.id`, so a similar field was selected (see below).

**Container**: No change in the number of documents with TSDB disabled v enabled
image

**CPU**: No change in the number of documents with TSDB disabled v enabled
image

**Diskio**: No change in the number of documents with TSDB disabled v enabled
image

**Event**: Does not have any metrics

**Healthcheck**: No change in the number of documents with TSDB disabled v enabled
image

**Image**: No change in the number of documents with TSDB disabled v enabled. Instead of `container.id`, we use `docker.image.id.current`.

image

Also changed:

image

To:

image

**Info**: `docker.info.id` instead of `container.id` is used (last one holds no value).
image

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] No change in the number of documents after enabling TSDB

## How to test this PR locally

1. Clone this branch.
2. Update the service registry to include this version of Docker and enable TSDB.

### CODE REVIEW

**Review by gizas — APPROVED** [review]

**Comment by constanca-m** [comment]
Hey @gizas , I added `agent.id` (dimension as well) to all metrics data streams. Reasoning can be found in the thread following this comment https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359

**Review by tetianakravchenko — COMMENTED** [review]
to align with https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359 I think that `host.name` should be added. I see that cloud.* are not defined in docker package, so lets skip it.

File: packages/docker/data_stream/image/fields/ecs.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
   name: ecs.version
 - external: ecs
   name: service.address
+  dimension: true
 - external: ecs
   name: service.type
 - external: ecs
```
Comment (tetianakravchenko) [inline packages/docker/data_stream/image/fields/ecs.yml:8]: `container.id` for `image` data_stream is it not set as a dimension on purpose?

Comment (constanca-m) [reply packages/docker/data_stream/image/fields/ecs.yml:8]: Yes @tetianakravchenko The field doesn't hold any value for that data stream, that's why `docker.image.id.current` is a dimension.

File: packages/docker/data_stream/info/fields/fields.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@
             Total number of existing containers.
     - name: id
       type: keyword
+      dimension: true
```
[the remark this answers is not in the document]
Comment (constanca-m) [reply packages/docker/data_stream/info/fields/fields.yml:30]: Exactly, `container.id` has no value for this data stream

**Comment by constanca-m** [comment]
> to align with [#5193 (comment)](https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359) I think that `host.name` should be added. I see that cloud.* are not defined in docker package, so lets skip it.

I think `service.id` should be enough for different hosts. What do you think @tetianakravchenko ?

**Comment by tetianakravchenko** [comment]
> > to align with https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359 I think that host.name should be added. I see that cloud.* are not defined in docker package, so lets skip it.
>
> I think service.address should be enough for different hosts. What do you think @tetianakravchenko ?

from my understanding there should be either the ip address or the socket, which might not be unique enough, if we run in different cloud providers/different cloud accounts instance with the docker service running, isn't it?

**Comment by constanca-m** [comment]
> from my understanding there should be either the ip address or the socket, which might not be unique enough, if we run in different cloud providers/different cloud accounts instance with the docker service running, isn't it?
> 
> in the sample file -
> 
> https://github.com/elastic/integrations/blob/c8e5b36e4461eb867307d9a6cfcc03f50f664b34/packages/docker/data_stream/info/sample_event.json#L24-L25
> 
> , so it might be the same for different instances, so we need to add some information about the host to separate data from different hosts

You're right. In my example they were different, but now that I think of it, they were only different because I made them be. I will fix it.

**Review by tetianakravchenko — APPROVED** [review]

**Pull request merged**